crates/provider/src/heart.rs

```diff
@@ -448,38 +463,121 @@ impl<S> Heartbeat<S> {
         }
     }
 
+    /// Reap transactions overridden by the reorg.
+    /// Accepts new chain height as an argument, and drops any subscriptions
+    /// that were received in blocks affected by the reorg (e.g. >= new_height).
```

Comment:
I think dropping reorged transactions isn't ideal, they become unconfirmed, because they can still be included in the new chain.

we could treat them as unconfirmed again after their block has been reorged.

Reply:
Sure, will do.

Reply:
Changed. Now transactions are returned to `self.unconfirmed`.